cicd/tekton-pipelines/cluster-config/templates/deploy.yaml

```diff
@@ -0,0 +1,156 @@
+apiVersion: tekton.dev/v1beta1
+kind: ClusterTask
+metadata:
+  name: deploy
+spec:
+  params:
+  - name: argo-app-name
+  - name: configGitUrl
+  - name: IMAGE
+  - name: ImageTag
+  - name: userHome
+    description: |
+      Absolute path to the user's home directory.
+    type: string
+    default: "/home/git"
+  - name: PARAM_SCM
+    description: Define the Source code Management URL
+    type: string
+    default: "github.com"
+  - name: userEmail  
+    description: "User email for Git configuration"
+    type: string
+  workspaces:
+  - name: config-source
+  - name: output
+    description: The git repo will be cloned onto the volume backing this Workspace.
+  - name: ssh-directory
+    optional: true
+    description: |
+      A .ssh directory with private key, known_hosts, config, etc. Copied to
+      the user's home before git commands are executed. Used to authenticate
+      with the git remote when performing the clone. Binding a Secret to this
+      Workspace is strongly recommended over other volume types.
+  - name: basic-auth
+    optional: true
+    description: |
+      A Workspace containing a .gitconfig and .git-credentials file. These
+      will be copied to the user's home before any git commands are run. Any
+      other files in this Workspace are ignored. It is strongly recommended
+      to use ssh-directory over basic-auth whenever possible and to bind a
+      Secret to this Workspace over other volume types.
+  - name: ssl-ca-directory
+    optional: true
+    description: |
+      A workspace containing CA certificates, this will be used by Git to
+      verify the peer with when fetching or pushing over HTTPS.
+  stepTemplate:
+    envFrom:
+    - secretRef:
```

Comment:
Can we take these from values.yaml? that means we need to create a secret which has admin privilege using this name?